prime/src/pages/BorrowActionsPage.tsx

```diff
@@ -415,6 +426,64 @@ export default function BorrowActionsPage() {
     [displayedMarginAccount, displayedUniswapPositions]
   );
 
+  // MARK: fetch uniswap fees earned
+  useEffect(() => {
+    let mounted = true;
+    if (marginAccount == null) {
+      return;
+    }
+    async function fetch(marginAccountLensContract: Contract, marginAccount: MarginAccount) {
+      const earnedFees: [string[], BigNumber[]] = await marginAccountLensContract.getUniswapFees(marginAccount.address);
+      const earnedFeesMap: UniswapPositionEarnedFees = {};
+      earnedFees[0].forEach((positionId, index) => {
+        earnedFeesMap[positionId] = {
+          token0FeesEarned: toBig(earnedFees[1][index * 2])
+            .div(10 ** marginAccount.token0.decimals)
+            .toNumber(),
+          token1FeesEarned: toBig(earnedFees[1][index * 2 + 1])
+            .div(10 ** marginAccount.token1.decimals)
+            .toNumber(),
+        };
+      });
+      if (mounted) {
+        setUniswapPositionEarnedFees(earnedFeesMap);
+      }
+    }
+    if (marginAccountLensContract && marginAccount) {
+      fetch(marginAccountLensContract, marginAccount);
+    }
+
+    return () => {
+      mounted = false;
+    };
+  }, [marginAccount, marginAccountLensContract]);
+
+  // MARK: compute uniswap fees earned
+  useEffect(() => {
+    if (!uniswapPositionEarnedFees || !displayedUniswapPositions || !marginAccount || !displayedMarginAccount) {
+      return;
+    }
+    const { token0, token1 } = marginAccount;
+    const earnedFeesValues = Object.values(uniswapPositionEarnedFees);
+    const priceX96 = marginAccount.sqrtPriceX96.mul(marginAccount.sqrtPriceX96).div(BIGQ96);
+    const token0FeesEarned = new Big(earnedFeesValues.reduce((p, c) => p.add(c.token0FeesEarned), new Big(0))).mul(
+      10 ** token0.decimals
+    );
+    const token1FeesEarned = new Big(earnedFeesValues.reduce((p, c) => p.add(c.token1FeesEarned), new Big(0))).mul(
+      10 ** token1.decimals
+    );
+    const token0FeesEarnedInTermsOfToken1 = priceX96.mul(token0FeesEarned).div(BIGQ96);
+    const totalFeesEarnedInTermsOfToken1 = token1FeesEarned.add(token0FeesEarnedInTermsOfToken1);
+    if (totalFeesEarnedInTermsOfToken1.gt(0)) {
+      if (isToken0Selected) {
+        const totalFeesEarnedInTermsOfToken0 = totalFeesEarnedInTermsOfToken1.mul(BIGQ96).div(priceX96);
+        setSwapFeesInputValue(totalFeesEarnedInTermsOfToken0.div(10 ** token0.decimals).toFixed(token0.decimals));
+      } else {
+        setSwapFeesInputValue(totalFeesEarnedInTermsOfToken1.div(10 ** token1.decimals).toFixed(token1.decimals));
```

Comment:
Same here?

Reply:
I did this to avoid showing more decimals than the token allows (since it could technically have more).